clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -2811,6 +2811,72 @@ public void testListConsumerGroupsWithStates() throws Exception {
         }
     }
 
+    @Test
+    public void testListConsumerGroupsWithTypes() throws Exception {
+        try (AdminClientUnitTestEnv env = new AdminClientUnitTestEnv(mockCluster(1, 0))) {
+            env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
+
+            // Test with a specific state filter but no type filter in list consumer group options.
+            env.kafkaClient().prepareResponse(prepareMetadataResponse(env.cluster(), Errors.NONE));
+
+            env.kafkaClient().prepareResponseFrom(
+                request -> request instanceof ListGroupsRequest &&
+                    !((ListGroupsRequest) request).data().statesFilter().isEmpty() &&
+                    ((ListGroupsRequest) request).data().typesFilter().isEmpty(),
```

Comment:
nit: It would be better to be precise here and validate if the states are actually the correct one. This also applies to the other matchers. `expectCreateTopicsRequestWithTopics` may be a good inspiration for the structure.

Reply:
haha okay, I actually started off by adding it to the unsupported version response method before getting the comments, it's honestly weird that the other tests don't validate anything lol, I had this urge to change the state tests also